lighthouse-core/gather/gatherers/web-app-manifest.js

```diff
@@ -94,8 +94,8 @@ class WebAppManifest extends FRGatherer {
    */
   getArtifact(context) {
     const driver = context.driver;
-
-    return WebAppManifest.getWebAppManifest(driver.defaultSession, context.url);
+    const {finalUrl} = context.baseArtifacts.URL;
+    return WebAppManifest.getWebAppManifest(driver.defaultSession, finalUrl);
```

Comment:
I was looking at the old spec, and it looks like "document URL" refers to `document.URL` which is affected by `history.pushState()` and such. I think we can safely use `finalUrl` in this case.